src/app/app.component.ts

```diff
@@ -40,24 +44,47 @@ export class AppComponent implements OnDestroy {
     }
 
     private processSourceChange(newSource: string): void {
-        const result = this._parserService.parse(newSource);
-        if (!result) {
-            return;
-        }
+        const errors = new Set<string>();
+        const result = this._parserService.parse(newSource, errors);
+        this.updateValidation(result, errors);
+
+        if (!result) return;
 
         this._displayService.updateCurrentRun(result);
     }
 
     private processNewSource(newSource: string): void {
-        const result = this._parserService.parse(newSource);
-        if (!result) {
-            return;
-        }
+        const errors = new Set<string>();
+        const result = this._parserService.parse(newSource, errors);
+        this.updateValidation(result, errors);
+
+        if (!result) return;
 
         this._displayService.registerRun(result);
         this.textareaFc.setValue(newSource, { emitEvent: false });
     }
 
+    private updateValidation(run: Run | null, errors: Set<string>): void {
+        this._runHint = [...errors, ...(run ? run.warnings : [])].join('\n');
+
+        if (!run || errors.size > 0) {
+            this.textareaFc.setErrors({ 'invalid run': true });
+            this._isRunValid = Valid.Error;
```

Comment:
_Nice to have_
Weiterführung von `enum` -> `type`

```suggestion
            this.runValidationStatus = 'error';
```